docs/frameworkevents.md

```diff
@@ -0,0 +1,27 @@
+# Framework events
```

Comment:
I wonder if it's better to link most of these events to this page instead: https://discourse.jujucharms.com/t/charm-hooks/1040

Reply:
So this raises a good point. One of the goals I wanted to address with this PR is the fragmented nature of docs.

It seems that docs for charms are here, and there. I was hoping for a single source of truth for all things writing charms.